File: vane-proxy-core/src/main/java/org/oddlama/vane/proxycore/listeners/PingEvent.java
package org.oddlama.vane.proxycore.listeners;

import org.oddlama.vane.proxycore.VaneProxyPlugin;
import org.oddlama.vane.proxycore.config.IVaneProxyServerInfo;

public abstract class PingEvent implements ProxyEvent {

    public VaneProxyPlugin plugin;
    public ProxyServerPing ping;
    public IVaneProxyServerInfo server;

    public PingEvent(VaneProxyPlugin plugin, ProxyServerPing ping, IVaneProxyServerInfo server) {
        this.plugin = plugin;
        this.ping = ping;
        this.server = server;
    }

    public void fire() {
        ping.setDescription(plugin.getMotd(server));
        ping.setFavicon(plugin.getFavicon(server));

        this.sendResponse();
    }

    public abstract void sendResponse();
}

File: vane-proxy-core/src/main/kotlin/org/oddlama/vane/proxycore/listeners/PingEvent.kt
package org.oddlama.vane.proxycore.listeners

import org.oddlama.vane.proxycore.VaneProxyPlugin
import org.oddlama.vane.proxycore.config.IVaneProxyServerInfo

/**
 * Base ping event that applies configured MOTD and favicon.
 *
 * @property plugin owning plugin instance.
 * @property ping mutable ping response wrapper.
 * @property server backend server currently being pinged.
 */
abstract class PingEvent(
    var plugin: VaneProxyPlugin,
    @JvmField var ping: ProxyServerPing,
    var server: IVaneProxyServerInfo
) :
    ProxyEvent {
    /** Applies presentation values and sends the ping response. */
    override fun fire() {
        ping.setDescription(plugin.getMotd(server))
        ping.setFavicon(plugin.getFavicon(server))

        sendResponse()
    }

    /** Dispatches the platform-specific ping response. */
    abstract fun sendResponse()
}
